database/schema/controller.go

```diff
@@ -56,6 +68,42 @@ ON lease_pin (lease_uuid, entity_id);
 CREATE INDEX idx_lease_pin_lease
 ON lease_pin (lease_uuid);
 `[1:]
+}
+
+func changeLogSchema() string {
+	return `
+CREATE TABLE change_log_type (
+    id   INT PRIMARY KEY,
+    type TEXT
+);
+
+CREATE UNIQUE INDEX idx_change_log_type_type
+ON change_log_type (type);
+
+INSERT INTO change_log_type VALUES
+    (1, 'create'),
+    (2, 'update'),
+    (4, 'delete');
+
+CREATE TABLE change_log_namespace (
+    id   INT PRIMARY KEY,
+    name TEXT
+);
+
+CREATE UNIQUE INDEX idx_change_log_namespace_name
+ON change_log_namespace (name);
 
-	return strings.Split(delta, ";\n\n")
+CREATE TABLE change_log (
+    uuid                    TEXT PRIMARY KEY,
+    change_log_type_id      INT NOT NULL,
+    change_log_namespace_id INT NOT NULL,
+    namespace_id            TEXT NOT NULL,
+    created_at              DATETIME NOT NULL,
```

Comment:
Might be worth adding `DEFAULT CURRENT_TIMESTAMP`, to save on setting explicitly for every trigger...

Reply:
Yeah

Reply:
Actually, I've changed this to use milliseconds now. This should give us better fidelity when grouping in the change stream.

Reply:
I think it's actually microseconds (2023-02-09 08:32:08.053).